src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2928,8 +3018,29 @@ private Socket getDefaultSocket(String hostName, int portNumber, int timeoutInMi
             InetSocketAddress cacheEntry = SQLServerConnection.getDNSEntry(hostName);
             addr = (null != cacheEntry) ? cacheEntry : addr;
         }
+
         return getConnectedSocket(addr, timeoutInMilliSeconds);
     }
+    
+    private ArrayList<InetAddress> getIPv6AddressList(InetAddress[] addresses) {
```

Comment:
Just to clarify, instead of using these getIPvXAddressList methods and creating new arraylist objects everytime, can't we just do one pass through of the InetAddresses and filter them once using member/instance arraylists?

For example:

private ArrayList IPv6AddressList;
private ArrayList IPv4AddressList;

Then in some appropriate place do the filtering like this:
       for (InetAddress addr : addresses) {
            if (addr instanceof Inet6Address) {
                IPv6AddressList.add(addr);
            } else {
                IPv4AddressList.add(addr);
        }

Reply:
This makes sense. Will adjust.